activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -30,7 +30,7 @@ def new_client(config)
           when ER_DBACCESS_DENIED_ERROR, ER_ACCESS_DENIED_ERROR
             raise ActiveRecord::DatabaseConnectionError.username_error(config[:username])
           when ER_CONN_HOST_ERROR, ER_UNKNOWN_HOST_ERROR
-            raise ActiveRecord::DatabaseConnectionError.hostname_error(config[:host])
+            raise ActiveRecord::DatabaseConnectionError.hostname_error("#{config[:host]}:#{config[:port]}")
```

Comment:
What if `config[:port]` is `nil`?

Reply:
That's a good point. I had assumed we would default the value based on the adapter but it doesn't look like we do (maybe the client libs still do? seems like another reason we should expose the client's error message as well).

I've updated the error to include `:port` if present, and made the `port` param optional so that other adapters don't break if they don't provide it.

Reply:
> maybe the client libs still do

Yes, at least for `pg` that is the case.  But that means if `conn_params[:port]` is `nil`, the error message would say `127.0.0.1` instead of `127.0.0.1:5432`.  So I feel like we should either manually include the default port, or omit the host entirely in favor of appending the client's error message.

Reply:
If we just replace `hostname` with `error.message`:

pg:
```
ActiveRecord::DatabaseConnectionError (There is an issue connecting with your hostname: connection to server at "127.0.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
.

Please check your database configuration and ensure there is a valid connection to your database.
):
```

mysql2:
```
ActiveRecord::DatabaseConnectionError (There is an issue connecting with your hostname: Can't connect to MySQL server on '127.0.0.1:3306' (111).

Please check your database configuration and ensure there is a valid connection to your database.
):
```

And trilogy only goes through this error if `TRILOGY_DNS_ERROR` is present in the message, which seems wrong to me but is probably out of scope for this change.

The `mysql2` error is somewhat interesting because it provides an error code (which was mentioned specifically in #50125 as something the user wanted to see), but it doesn't look like the pg adapter gives us anything interesting.

Do these seem reasonable to just always include? Alternatively, we could include the MySQL message but write a custom one for postgres using default config values that's a bit more succinct?